soccer/src/soccer/strategy/agent/agent_action_client.cpp

```diff
@@ -75,6 +80,34 @@ void AgentActionClient::get_task() {
     }
 }
 
+void AgentActionClient::update_position(const rj_msgs::msg::Position::SharedPtr& msg) {
+    // TODO remove this debug
+    /* for (int i = 0; i < 6; i++) { */
+    /*     SPDLOG_INFO("position at {}: {}", i, msg->client_positions.at(i)); */
+    /* } */
+    // SPDLOG_INFO("{}'s position : {}", robot_id_, msg->client_positions[robot_id_]);
+    if (robot_id_ == 0) {
+        return;
+    }
+
+    std::unique_ptr<Position> next_position_;
+    switch (msg->client_positions[robot_id_]) {
+        case 1:
+            next_position_ = std::make_unique<Defense>(robot_id_);
+            break;
+        case 2:
+            next_position_ = std::make_unique<Offense>(robot_id_);
+            break;
+    };
+
+    if (current_position_ == nullptr ||
+        next_position_->get_name() != current_position_->get_name()) {
+        current_position_ = std::move(next_position_);
+    }
+
+    // TODO: send acknowledgement back to coach node (w/ robot ID)
```

Comment:
can you send an ACK in this PR, and also cache (save) the latest acknowledged positions in the coach node (think of it as the Coach has authority over position assignments, and so should know what all positions are doing)

Reply:
Will be a new PR. Coach will sub to an ACK topic from AC and save the state of the acknowledgements.